python/cuml/cuml/solvers/sgd.pyx

```diff
@@ -202,294 +344,145 @@ class SGD(Base,
-                                                  else None),
-                                check_cols=self.n_cols)
-
-        cdef uintptr_t _X_ptr = X_m.ptr
+        cdef int n_rows, n_cols
+        X, n_rows, n_cols, _ = input_to_cuml_array(
+            X,
+            check_dtype=self.coef_.dtype,
+            convert_to_dtype=(self.coef_.dtype if convert_dtype else None),
+            check_cols=self.coef_.shape[0],
+        )
 
-        cdef uintptr_t _coef_ptr = self.coef_.ptr
-        preds = CumlArray.zeros(_n_rows, dtype=self.dtype, index=X_m.index)
-        cdef uintptr_t _preds_ptr = preds.ptr
+        preds = CumlArray.zeros(n_rows, dtype=self.coef_.dtype, index=X.index)
 
         cdef handle_t* handle_ = <handle_t*><size_t>self.handle.getHandle()
-
-        if self.dtype == np.float32:
-            sgdPredict(handle_[0],
-                       <float*>_X_ptr,
-                       <int>_n_rows,
-                       <int>_n_cols,
-                       <float*>_coef_ptr,
-                       <float>self.intercept_,
-                       <float*>_preds_ptr,
-                       <int>self._get_loss_int())
-        else:
-            sgdPredict(handle_[0],
-                       <double*>_X_ptr,
-                       <int>_n_rows,
-                       <int>_n_cols,
-                       <double*>_coef_ptr,
-                       <double>self.intercept_,
-                       <double*>_preds_ptr,
-                       <int>self._get_loss_int())
-
+        cdef int loss_code = _LOSSES[self.loss]
+        cdef bool use_f32 = self.coef_.dtype == np.float32
+        cdef uintptr_t preds_ptr = preds.ptr
+        cdef uintptr_t X_ptr = X.ptr
+        cdef uintptr_t coef_ptr = self.coef_.ptr
+        cdef double intercept = self.intercept_
+
+        with nogil:
+            if use_f32:
+                sgdPredict(
+                    handle_[0],
+                    <float*>X_ptr,
+                    n_rows,
+                    n_cols,
+                    <float*>coef_ptr,
+                    intercept,
+                    <float*>preds_ptr,
+                    loss_code,
+                )
+            else:
+                sgdPredict(
+                    handle_[0],
+                    <double*>X_ptr,
+                    n_rows,
+                    n_cols,
+                    <double*>coef_ptr,
+                    intercept,
+                    <double*>preds_ptr,
+                    loss_code,
+                )
         self.handle.sync()
 
-        del X_m
-
         return preds
-
-    @generate_docstring(return_values={'name': 'preds',
-                                       'type': 'dense',
-                                       'description': 'Predicted values',
-                                       'shape': '(n_samples, 1)'})
-    @cuml.internals.api_base_return_array(get_output_dtype=True)
-    def predictClass(self, X, convert_dtype=True) -> CumlArray:
```

Comment:
I would prefer to keep the function stub and raise an exception with a meaningful error message that describes what a user should use/do instead as opposed to completely removing it.